packages/mask/src/plugins/Referral/helpers/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { padStart } from 'lodash-unified'
+import BigNumber from 'bignumber.js'
+import { createTypedMessageMetadataReader } from '@masknet/typed-message/dom'
+import { formatUnits } from '@ethersproject/units'
+import { isValidAddress } from '@masknet/web3-shared-evm'
+
+import { META_KEY, supportedChainId } from '../constants'
+import type {
+    ReferralMetaData,
+    ChainAddress,
+    ChainAddressProps,
+    EvmAddress,
+    Bytes32,
+    Bytes24,
+    ChainId,
+    BigNumberish,
+} from '../types'
+import schema from '../schema.json'
+import { bufToHexString, toBigInt, writeUInt32BE, hexToArrayBuffer } from './buffer'
+
+function toChainAddress(chainId: BigNumber | bigint, address: Uint8Array): Uint8Array {
+    if (isValidAddress(address.toString())) throw new Error('invalid address')
```

Comment:
Learn more: https://web3js.readthedocs.io/en/v1.7.3/web3-utils.html#bytestohex

```suggestion
    if (isValidAddress(bytesToHex(address))) throw new Error('invalid address')
```

Reply:
`bytesToHex(bytes: number[]): string`, address here is` address: Uint8Array`.
TS error: `Argument of type 'Uint8Array' is not assignable to parameter of type 'number[]'.`

Probably do you know some utils to convert Uint8Array to string?

Reply:
It can expand to an array of numbers. I noticed that the `0x` should add as a prefix.

```ts
`0x${bufToHexString(...address)}`
```

Reply:
Yes, thanks, done refactoring it and [bytestohex](https://web3js.readthedocs.io/en/v1.7.3/web3-utils.html#bytestohex)